scripts/fuzz_opt.py

```diff
@@ -103,6 +146,61 @@ def compare(x, y, context):
         ))
 
 
+# numbers are "close enough" if they just differ in printing, as different
+# vms may print at different precision levels and verbosity
+def numbers_are_close_enough(x, y):
+    # handle nan comparisons like -nan:0x7ffff0 vs NaN, ignoring the bits
+    if 'nan' in x.lower() and 'nan' in y.lower():
+        return True
+    # float() on the strings will handle many minor differences, like
+    # float('1.0') == float('1') , float('inf') == float('Infinity'), etc.
+    try:
+        return float(x) == float(y)
+    except Exception:
+        pass
+    # otherwise, try a full eval which can handle i64s too
+    try:
+        x = eval(x)
+        y = eval(y)
+        return x == y or float(x) == float(y)
```

Comment:
Could the `float(x) == float(y)` here give a false positive for two very close large i64s?

Reply:
I think that's possible, yes.

To really handle this properly we'd need to know the type of each logged value, which the current notation doesn't support.

Reply:
alright, doesn't seem like it will be a huge problem in practice.